cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,243 @@
+"""
+Tests for the maintenance app views.
+"""
+import ddt
+
+from django.core.urlresolvers import reverse
+from student.tests.factories import AdminFactory, UserFactory
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+from contentstore.management.commands.utils import get_course_versions
+
+from .views import COURSE_KEY_ERROR_MESSAGES, get_maintenace_urls
+
+
+class TestMaintenanceIndex(SharedModuleStoreTestCase):
+    """
+    Tests for maintenance index view.
+    """
+
+    def setUp(self):
+        """Make the user global staff. """
+        super(TestMaintenanceIndex, self).setUp()
+        self.user = AdminFactory()
+        login_success = self.client.login(username=self.user.username, password='test')
+        self.assertTrue(login_success)
+        self.view_url = reverse('maintenance:maintenance')
+
+    def test_maintenance_index(self):
+        response = self.client.get(self.view_url)
+        self.assertContains(response, "Maintenance", status_code=200)
+
+        # Check that all the expected links appear on the index page.
+        for url in get_maintenace_urls():
+            self.assertContains(response, url, status_code=200)
+
+
+@ddt.ddt
+class MaintenanceViewTestCase(SharedModuleStoreTestCase):
```

Comment:
@mushtaqak , I'm not sure you're using the using the `SharedModuleStoreTestCase` correctly. The idea is that you set up your course in a `setUpClass` method that's a classmethod (so that it's only set up once per test class). Here's Dave's original blog post: http://swampcastle.org/2015/07/28/writing-faster-modulestore-tests.html

I'm happy to walk you through this tomorrow morning if that would be helpful!